micropip/package_index.py

```diff
@@ -0,0 +1,121 @@
+from collections import defaultdict
+from dataclasses import dataclass
+from typing import Any
+
+from packaging.utils import (
+    parse_wheel_filename,
+)
+from packaging.version import InvalidVersion, Version
+
+
+@dataclass
+class ProjectInfoFile:
```

Comment:
`WheelInfo` maybe? That's a similar container to https://github.com/pyodide/pyodide/blob/main/pyodide-build/pyodide_build/mkpkg.py#L23 BTW might we worth mentioning it in a comment.

Reply:
We are already using that name here.

https://github.com/pyodide/micropip/blob/501a6910e79181b5a9ccfefa2ec88728a99e04d6/micropip/transaction.py#L37

Maybe we can refactor these two classes in a follow-up, by the way. They contain similar attributes.

Reply:
Ahh yes. Maybe make this `BaseWheelInfo` and inherit from it there, since it's a subset of those fields?

Reply:
> We can also do it later, but someone would need to remember it. I probably wont :)

This (Simple API and related stuff) is my high-priority work for now, so I'll remember. I'll add some comments so I won't forget :)